### DISCUSSION TITLE
Easy issue creation with TODOs

### DISCUSSION THREAD

**Discussion by drkameleon** [h3rald/min#155] [Ideas]
Since - apart from working on my project - I've been watching yours too, here's an idea I came across, which you might also like:

I was thinking of somehow **documenting (for myself, first) what needs to be done**, what's missing, etc. 

So, I thought: _why not add  TODOs IN my code?_

Well, that's an obvious one. But now, how do I track them? I mean, it's nice to have them around, but hidden?!

Then, I discovered **Todo Tree**: https://marketplace.visualstudio.com/items?itemName=Gruntfuggly.todo-tree (since I've been experimenting with VSCode lately, the plugin works great, with all the TODOs in a nice tree, by file).

So far so good.

But then... I wanted more. :) What if we were able to automatically **convert those TODOs into issues**, and have them "closed"/resolved once the relevant TODO annotation goes away?

So...

- Here's what I came up with: https://github.com/arturo-lang/arturo/blob/master/.github/workflows/todos.yml (it should work pretty much instantly - let me know in case you wouldn't mind a PR).

- That's what an "issue" looks like in the code: https://github.com/arturo-lang/arturo/blob/master/src/library/Core.nim#L34

- And this is the result: https://github.com/arturo-lang/arturo/issues

Looks like a complete mess at first sight, but it definitely helps me know what is pending - and what my progress is. (+ it gives a hopefully good incentive to people to contribute to the project - still waiting to see that happen though lol)

Just a few thoughts... Have a great day! :)

**Comment by h3rald** [comment]
Ohhh that's interesting! I am always unsure on whether to create "feature" issues in advance with my OSS projects or not. 

At the moment I am basically the sole contributor of my projects for the most part, and I have successfully been using private Trello Kanban boards to keep track of things. The reason being that I can quickly move things around in different columns, add check lists etc.

Now, what I would *really* like is for GitHub to integrate their Projects feature into their mobile app! When they do that, I'll probably switch to it and leave Trello.

Another problem is also that at the moment my free time is really limited, and it is rare that my newborn daughter allows me the luxury to fire up VSCode (which I use at work everyday) to code... most of the things I implemented in min and other projects over the past few months have actually been implemented... on my iPhone or iPad 😬 at stupid hours of the night (after I change her nappy, typically I can't sleep right away so sometimes I code).

I use a combination of Termius to connect to my Mac in the other room or one of my Linux servers/shells via ssh and the excellent [buffer editor](https://buffereditor.com), which has vim-like bindings and barely provides syntax highlighting (I have to use Python highlighting for Nim files 🤣), but hey, it does the job!
